pos-print/src/main/java/com/clouway/pos/print/adapter/http/ReportService.kt

```diff
@@ -58,17 +57,24 @@ class ReportService(private var factory: PrinterFactory) {
     return Reply.with("Report completed").`as`(Json::class.java).ok()
   }
 
-  private fun adapt(state: String): RegisterState {
+  private fun adaptToState(state: String): RegisterState {
     if (state.equals("CLEAR")) {
       return RegisterState.CLEAR
     }
     return RegisterState.KEEP
   }
 
-  private fun adaptToDate(date: String): LocalDate {
-    return LocalDate.parse(date, DateTimeFormatter.ISO_DATE_TIME)
+  private fun adaptToType(type: String): PeriodType {
+    if (type.equals("SHORT")) {
+      return PeriodType.SHORT
+    }
+    return PeriodType.EXTENDED
+  }
+
+  private fun adaptToDate(date: String): LocalDateTime {
+    return LocalDateTime.parse(date, DateTimeFormatter.ISO_DATE_TIME)
```

Comment:
also if you make the gson parser work, then you will have to pass dates as dto's properties(not string), then passing the json you will have and an exaple of the required message. This will efficiently test and the date parser too.